### DISCUSSION TITLE
npm i node-fetch, npm ERR! code ETIMEDOUT

### DISCUSSION THREAD

**Discussion by VardanMelik** [node-fetch/node-fetch#1766] [Q&A] [answered]
Hi guys, 
I have an issue during package installation. 

npm i node-fetch
npm ERR! code ETIMEDOUT
npm ERR! syscall connect
npm ERR! errno ETIMEDOUT
npm ERR! network request to https://registry.npmjs.org/data-uri-to-buffer failed, reason: connect ETIMEDOUT 2606:4700::6810:1d22:443
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'

OS: Windows 11
Node: v18.17.0
NPM: 9.6.7

What I did to solve, but still have this issue
- Removed/Installed Node.js

Can some one advice how to fix?
Thx

**Comment by VardanMelik** [comment]
After changing Node.js version:

Node.js: v20.5.0
NPM:     9.8.0

"node-fetch": "^3.3.2",

const fetch = require("node-fetch");
              ^

Error [ERR_REQUIRE_ESM]: require() of ES Module

But this code worked with previous versions.

**Comment by VardanMelik** [comment, marked as answer]
After installing 

npm i node-fetch@2.6.7
Node.js: v20.5.0
NPM: 9.8.0

This work for me. May be this can be useful for someone.

**Reply by Snail110** [reply]
thanks, I upgraded my node version,then solved the problem
